indirect-connection/main.py

```diff
@@ -0,0 +1,61 @@
+#!/usr/bin/env python3
+
+from locust import task
+from locust.contrib.fasthttp import FastHttpUser
+
+import os
+
+class AdmissionRequest:
+    def __init__(self, endpoint, request_file, expected_response_http_status):
+        with open("/var/run/secrets/kubernetes.io/serviceaccount/token", "r") as f:
+            self._auth_token = f.read()
+
+        self._endpoint = endpoint
+
+        self._expected_response_http_status = expected_response_http_status
+
+        req_file_dir = os.environ.get("LOCUST_TEMPLATES_DIR", "lib")
+        with open(os.path.join(req_file_dir, request_file), "r") as f:
+            self._request = f.read()
+
+    @property
+    def endpoint(self):
+        return self._endpoint
+
+    @property
+    def expected_response_http_status(self):
+        return self._expected_response_http_status
+
+    @property
+    def data(self):
+        return self._request
+
+    def eval(self, user):
+        headers = {
+            "Content-Type": "application/json",
+            "Accept": "application/json",
+            "Authorization": "Bearer {}".format(self._auth_token),
+        }
+
+        with user.client.post(
+                self.endpoint,
+                data=self.data,
+                headers=headers,
+                catch_response=True) as response:
+            if response.status_code == self._expected_response_http_status:
+                response.success()
+            else:
+                response.failure("Wron response code: got {} instead of {}".format(
+                    response.status_code,
+                    self._expected_response_http_status))
+
+class MyUser(FastHttpUser):
+    tasks = [
+        AdmissionRequest(
+            endpoint="/apis/apps/v1/namespaces/default/deployments",
+            request_file="nginx-echo.json",
+            expected_response_http_status = 418,
```

Comment:
My indirect tests are still failing. I think we need to change the expected http status. I'm quite surprise that the [418 status code](https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/418) is used. I've tried to check the swagger definition of the API endpoint to see the expected return status code. But I could find the 418.

Considering the status code that I'm getting in the my local tests I think we need to change the name of the resource being deployed or update the deployment instead of creating it every time. The impression that I get from the errors from the image bellow is that the first time the resource has been created fine and the following request trying to create it again failed because it is already created and we can only updated it. But I'm not 100% if my hypothesis is right.  
![image](https://user-images.githubusercontent.com/1514798/180224994-54755aa7-0572-4379-9e88-5941405adba6.png)

Additional links:
https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/409

Reply:
(I can reproduce the indirect tests failing here too)

Reply:
I picked the teapot http status on purpose, this is returned by the echo policy.

Are you sure you don't have some leftovers inside of the `default` namespace? Have you updated the configmap holding the source code of the unit tests and restarted locust?

I tried it against a freshly provisioned cluster and it worked :thinking:

Reply:
It keeps showing the same wrong behaviour here, always on a fresh cluster and with the correct configmaps for the indirect tests.

Here, the echo policy never kicks in and rejects and logs (as evidenced by the http response codes 201, 409). Maybe misconfigured RBAC?

Reply:
I've applied your suggestion. Can I mark this comment as resolved and merge the PR?

Reply:
BTW, thanks for the fix!